### ISSUE TITLE
Allow complex asset URIs

### ISSUE THREAD

**Issue by basxto** [openfl/openfl#1872]
Works:
```
var path = "assets/data/terrains/dirt.tsx"
if (Assets.exists(path))
{
  trace(Assets.getText(path))
}
```

Doesn't, but should:
```
var path = "assets/data/maps/../terrains/dirt.tsx"
if (Assets.exists(path))
{
  trace(Assets.getText(path))
}
```

**Comment by tienery** [comment]
I believe Assets in OpenFL, like in Kha, generates the Assets based on a macro so that it can be easily accessible. The path is therefore hardcoded.

 

You can use Bitmap.FromFile(path) which would work as just in your second example code, but it would only work on targets that access the File System.

 

Hope that helps.

**Comment by Gama11** [comment]
As @tienery said, it's less of a path and more of an asset ID (which just happens to be a path by default).

**Comment by ibilon** [comment]
For that specific case you can use `haxe.io.Path.normalize(path)`.

**Comment by basxto** [comment]
Okay, thank you all for your quick replies.

@ibilon I was looking for such a function, but could not find it.

**Comment by jgranick** [comment]
You might be looking for:

```haxe
#if sys
var path = "assets/data/maps/../terrains/dirt.tsx"
if (sys.FileSystem.exists (path)) {
    trace (sys.io.File.getContent (path));
}
#end
```

**Comment by basxto** [comment]
The code was an excerpt of [flixel.addons.editors.tiled.TiledTileSet](https://api.haxeflixel.com/flixel/addons/editors/tiled/TiledTileSet.html), therefore I was looking for an as platform independent solution as possible.

This situation occurs, when you have assets (map file), which reference other assets (tilesets) with relative paths. `assets/data/maps/` + `../terrains/dirt.tsx`

Maybe it would be better to call it „primitive asset path“ or „simplified asset path“ in the API documentation? Or maybe mention in the general description of openfl.utils.Assets, that paths must be normalized.

**Comment by Gama11** [comment]
Yeah, it might be good to clarify that it's not really a "file system path", just an ID that happens to be one.
